routes/user.js

```diff
@@ -0,0 +1,237 @@
+      res.send({
+        code    : 400,
+        message : 'Email must be between 4 and 512 characters long.'
+      });
+      return;
+    }
+
+    User.create(username, password, email, function(err, session) {
+      if (err) {
+        res.send(err);
+        return;
+      }
+      res.send({
+        code    : 200,
+        message : 'Account successfully created.',
+        token   : session.token,
+        ttl     : session.expireAt - Date.now()
+      });
+    });
+  }
+
+  function deleteAccount(req, res) {
+    var clientToken = req.body.token;
+
+    User.delete(clientToken, function(err) {
+      if (err) {
+        res.send(err);
+        return;
+      }
+      res.send({
+        code    : 200,
+        message : 'Account successfully deleted.'
+      });
+    });
+  }
+
+  function changePassword(req, res) {
+    var clientToken = req.body.token;
+    var oldPassword = validator.toString(validator.escape(req.body.oldPassword));
+    var newPassword = validator.toString(validator.escape(req.body.newPassword));
+
+    User.changePassword(clientToken, oldPassword, newPassword, function(err) {
+      if (err) {
+        res.send(err);
+        return;
+      }
+      res.send({
+        code    : 200,
+        message : 'Password has been changed.'
+      });
+    });
+  }
+
+  function authenticate(req, res) {
+    var usernameEmail = validator.toString(validator.escape(req.body.usernameEmail));
+    var password      = validator.toString(validator.escape(req.body.password));
+
+    if (!usernameEmail) {
+      res.send({
+        code    : 400,
+        message : 'Username/Email field is required.'
+      });
+    }
+
+    // TODO: Decide on min and max
+    // Check usernameEmail length
+    if (!validator.isLength(usernameEmail, 4, 512)) {
+      res.send({
+        code    : 400,
+        message : 'Username/Email must be between 5 and 512 characters.'
+      });
+      return;
+    }
+
+    if (!password) {
+      res.send({
+        code    : 400,
+        message : 'Password field is required.'
+      });
+      return;
+    }
+
+    // TODO: Decide on min and max
+    // Check password length
+    if (!validator.isLength(password, 4, 512)) {
+      res.send({
+        code    : 400,
+        message : 'Password must be between 5 and 512 characters.'
+      });
+      return;
+    }
+
+    User.login(usernameEmail, password, function(err, session) {
+      if (err) {
+        res.send(err);
+        return;
+      }
+      res.send({
+        code    : 200,
+        message : 'Login successful.',
+        token   : session.token,
+        ttl     : session.expireAt - Date.now()
+      });
+    });
+  }
+
+  function reauthenticate(req, res) {
+    var clientToken = req.body.token;
+    User.reauthenticate(clientToken, function(err, session) {
+      if (err) {
+        res.send(err);
+        return;
+      }
+      res.send({
+        code    : 200,
+        message : 'Login successful.',
```

Comment:
Change this message to 'Reauthentication Successful', or whatever you think would be appropriate.